samples/outreach/blogs/segmentation_blogpost/5_Image_Segmentation.ipynb

```diff
@@ -0,0 +1,1840 @@
kWfHtuOOOS71GSeEqeY2y9idJ1jspPCQ1J+H3pD1OXDOSlm+climflOsX5he/j/H7knbdw/sRXtf4/Yrb+oRCU7y8NA1Y/DqHn0Ot2AMPPMDk9pf4jTe/Fbs03pRXsuws2kWXNhgMhqVEx5FeRL4qIltF5KexfTeKyBMi8piI/J2ITMSObRCRp0XkSRH5ndj+dwX7nhaRC7qpXHxqoN2bftqbaNyoNSRr8AuPpaVNToNYVrOBcRaRTUswUO7cuZPJycnIgwWy49nkcjmcXCMYYDcCmbbJcBBy2FZBe9yo5nq2EwBCt+J4eU1plaNdwcUFcZu0HsmprDB95EWWIfhkCTbhsai+kgfloHxb/48LWrFy0q5Jk2GzcqK8ws2SPJbkWwTPtPqGOI7TZIScj7mcg9YS+V62UXQTsTo0Gu/y+M8eZduLmzn6dUfi5wugPITW30DT8gyGVOazDzMYDHNLN+qLvwDeldj3beAopdTrgV8AGwBE5EjgJOC1wTl/JiK2iNjAF4B3A0cCfxikbYuv/EjjEvcuCT+HW6ilSL4Rx208klqbpLAS399Sj8SAkTbopWlywjgrhUKBNWvWMD4+TrFYbNFGpLbdsyLj4jjJtrfT3oSGsUnX42hTNVyvgudX8fxqehpPCw6uV0n1IAvrGl2HYICOBCBaNU3xaxbeo/A8xy7i2EX9PdDA1d1yYOTs6nwTgkxTuxPCVJMGxLbwRP9PajyEXCRYxIWpUAiK2pUhJHoSa1POwROiLRTMw+ci+Uy6rksuX2RYjbL56Z+zbfsLvObIVzGxcjVWvhQJqlEdJI9tFbCtAnXVGgMnSb/G5kuIv2Ce+jCDwTC3dBRqlFLfA3Yk9v2TUir0Bb4PWBt8PhG4WylVVUptBJ4G3hhsTyulnlFK1YC7g7Rtsa0cpVIpGsjj0zqdtBfx1bLb2eOk5ZOltUk7ljZghOmLxSLbtm3TA3O9nlnXdoJJGFyv08AVpo3nFzfCjf9Plr1ly5am4IDx/JKB4+Jaqvjx+PdkWWFdurWtiRO3SenU7jSS9yd5reNTT2nXOO3ehO3IEpRD4TvcD+lLfMTj8vz04ft4Ye9POeSIV3PA6leQs0ciF++40B5OJ+bzeRzHiVYdb9f+fX2phPnswwwGw9wyCJuaPwH+T/D5QOD52LFNwb6s/S2IyHoReVBEHty1azv/++++haeGkVwenAKSK6LsfItXUzIGSaglieUb/U966WSRNW0S/66UioK7xQe+oaEhKpUKq1ev7quc+ODYSRCICwtxASIr/4hAA/HKQ38tmqqKyg00JtHgHNNaRFswFRUeb3Zxb2g2kvYiqcTyitcP5VDID1Mpu1pzpZpd0H1V01NhbaahQmzV2EJCbUpTvcKpqZgQE+URs++J23rZKc2KX/fJyUk8z8MTcFGI2IwMrebRH30Px9/B4Ue+npGRA6i5dWr1CnW3St0tU3fLjTXJIIo11BAe2wu7WTF/DE3MWh82C3U1GAxtmJFQIyIXAS7wjXBXSjLVZn/rTqXuUkodq5Q6dnxiGb/9e+9D/O1M7niZjT/7eYtAEnbo8bfZNvXtql3xfLtJF18E0vd9JicnmZqa6jqPmdhDpE2NdTOFFt/fyWYpjTTNUtb1T3oIpeXT6d4Ui0XK5XJmPfohaTyczDNNs5R0349PLbUrZ2xsTNvf+LD7pZd58fmnqFWe43W/cRw1xqL8kmVAugasnfF8kn1dU9OO2e7DBlNLg8HQLX27dIvIR4H3AG9XjV5zE3BQLNlaYEvwOWt/dhmA5ddRkmdkNM+yFRNMT+/gX/75e5xwwu9Tq3qQAxy/5SUfsmPN9Eu9Xo+EpzQqlQqe5zE6Ogok1jwKusS0N/qsmDBpmoJkuvj/ToTLOsQ1GqG3VXyJB9/3G+tDqYYXT4vg0UEzEqbR9Y/VMbxZgbanYeRMY1mIMGmszGKxmLr8Qxrd3ntbJbzdEga7YV7gZ5aV5TIduWaLUKvVeObnP+HXjjyCZfuNYlurcREsAcsOr28eXzUe5NCo3MdrWrIjfDY8z8OmBlYxpXRDJ+aiDzMYDHNLX6+4IvIu4Hzg95VS07FD9wIniUhBRA4FDgd+BDwAHC4ih4pIHm2Id283ZeVyOWzbxnEcahWbQn6Cd7zj3YBPvjTFI/d9m23PlKPw+XH6jaqaNRiGKzMnKRQKFItFisUio6OjM9IeJAfjdgNz/O09K13cDimM+RLnySefjJZnSLteafYjvZJmJJw8Hq9v+D9NqxM3ru2m3Hi+yQ3aX7ukxiY8L3QzD7e4m3m45fN5nnviKTY+/gT+9A7y/jRHHXkMlgxjyXBT+a7rRpqXuB1NPp/Htu0mQTpp29NNDBoz/dTKXPZhBoNh7uj4qi0i/x14C7BSRDYBl6E9BQrAt4MO8z6l1J8qpR4XkW8CP0OrdE9VSnlBPqcB/4iOIf9VpdTjncpWCHVlQdiJ2xU8X4EdvB17JY75zbeipMKuPZt5+qnnwBvjmGOP1m+5hcDupeaC1XBBDgexsLNPqudDQaCTYGJZFtu2bWP16tXU6/UWA91uDHuT0x5J25hOGoekVieZf/Kz7/vYYVJxOe74Y9n28q7WMkKbEstvW0Yul8s0gk47J9J6BOU36tYsgKRphiJtF43lDbwO47UfOx7XkqngIsQ1NS3ti2mZwjoly43OVw4jMsWjjz7KqrWHMDExwZpD11AqlSLBw5NaNIfhh+dJcH29xjScb+lpsVzMHknRiOacvDZxTVscYySsmc8+zGAwzC2ykDu9I179GnXXV/4bkP5GrQfNYL/rNXnq+L7PP/793/C77/5DXIZQVPSAwR59rpeudUnTSiTLDd/Q07QF3Ux5JNO0m5rqRJbgFM8zjNcSCWu2T71ejxbRLBVH2+bfjYDXrm7t7Hna7U9O3SXrVK/XGcoXtMaiT2VEUqAJ74GItN6XvEVeCnraR7TA8+P7/i+VSoU3HP9b5HK55jXBEnOiTbZgQWwcN5huy4l
```

Comment:
The dice loss is [almost] just a differentiable IOU... right?

Reply:
Right, they are almost the same, but I think there are differences when quantifying the the differences in performance over the entire set of predictions. I think dice tends to measure something closer to average performance, while the iou score measures something closer to the worst case performance.

Honestly, it's probably clearer for the user if we just remove the iou metric, I initially included it to illustrate how to add different custom metrics and give examples of how to implement them.